python/cuml/ensemble/randomforestregressor.pyx

```diff
@@ -669,7 +673,7 @@ class RandomForestRegressor(Base):
                                              storage_type=storage_type)
 
         preds = tl_to_fil_model.predict(X, out_type)
-        tl.free_treelite_model(treelite_handle)
+        TreeliteModel.free_treelite_model(treelite_handle)
```

Comment:
Why is the treelite model freed here but not freed in case of classification?

Reply:
Basically the new approach (in classifier) that uses obtain_treelite_handle everywhere allows the destructor to free the models, while the old approach of allocating locally within this function required a local cleanup. I think the obtain_treelite_handle approach is better as you noted above.